packages/tools/pluggable-widgets-tools/configs/rollup.config.js

```diff
@@ -222,6 +223,15 @@ export default async args => {
     function getClientComponentPlugins() {
         return [
             isTypescript ? widgetTyping({ sourceDir: join(sourcePath, "src") }) : null,
+            isTypescript
```

Comment:
Does it worth to re-format all the project for every change when running `npm start`?

Reply:
It might slow down a lot the build/release step

Reply:
Yes, because the widget properties defined in the xml can change when `npm start` is running. Basically every time, the widget props are generated again, you want to format it according to prettier.

Reply:
👍🏻

Reply:
Do we really need to run Prettier for the typings? Or, to phrase it differently, in what way are the generated typings currently "unpretty"? I'm a bit worried about adding more and more steps slowing down the tooling, so if we can prevent them from being necessary in the first place, that might be a better option.

Reply:
@tamelbrun Good question. What I often encounter during development on widgets is that long enums are generated on a single line. Prettier chooses to give new lines for each enum item. Running the build script without changing the properties in the xml would give git changes in the prop typings, which I think isn't intuitive.

In the monorepo a pre-commit hook will prettify the typings in the end, so that no changes are committed for the typings. However, pre-commit hook execution sometimes needs special Husky configuration on a developers local machine in order to work, which increases the risk of committing code style changes.

Widget projects generated by customers don't have pre-commit hooks at all, so they have an even higher chance of committing code style changes. I think we should help customers prevent polluting commit history as much as possible and, therefore, limit these code style change commits.

Reply:
Ah, I agree that it would be too much work to mimic Prettier's behaviour when generating typings, when Prettier can do that for us. But we're only prettifying the typings now, right? Because then there's probably not much of a performance hit from doing this.

I'm curious though... if customers don't have these pre-commit hooks, do they still have Prettier configured? And when is it run? Because there will only be unintentional code style changes if Prettier (or another formatting tool) is actually running at an unexpected moment. So I'm still not entirely sure which exact scenario we're solving here.

Reply:
I was thinking about something else: if Prettifying the typings is deemed necessary, shouldn't that be part of the `widgetTyping` plugin itself? That way, the behaviour of generating typings is the same for web and native.

Reply:
> But we're only prettifying the typings now, right? Because then there's probably not much of a performance hit from doing this.

Correct.

> I'm curious though... if customers don't have these pre-commit hooks, do they still have Prettier configured? And when is it run?

Prettier is installed via `npm i`. For customer generated projects prettier is run when executing the `npm run lint(:fix)` command. Customers don't have pre-commit hooks by default, so unintentional style code changes can still occur. We can start shipping pre-commit hooks to improve the experience even more by using [this library](https://www.npmjs.com/package/pre-commit).

> I was thinking about something else: if Prettifying the typings is deemed necessary, shouldn't that be part of the widgetTyping plugin itself? 

I think it should be part of the plugin indeed.